src/lyrics/lyrics.cc

```diff
@@ -0,0 +1,330 @@
+ * Copyright (c) 2024 Thomas Lange <thomas-lange2@gmx.de>
+ *
+ * Permission to use, copy, modify, and/or distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE AUTHOR ``AS IS'' AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
+ * IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <json-glib/json-glib.h>
+
+#include <libaudcore/drct.h>
+#include <libaudcore/hook.h>
+#include <libaudcore/plugin.h>
+#include <libaudcore/plugins.h>
+#include <libaudgui/gtk-compat.h>
+
+#include "lyrics.h"
+#include "preferences.h"
+
+class Lyrics : public GeneralPlugin
+{
+public:
+    static const char * const defaults[];
+    static const PluginPreferences prefs;
+    static const PreferencesWidget widgets[];
+    static constexpr PluginInfo info = {
+        N_("Lyrics"),
+        PACKAGE,
+        nullptr, // about
+        & prefs,
+        PluginGLibOnly
+    };
+
+    constexpr Lyrics () : GeneralPlugin (info, false) {}
+
+    bool init ();
+    void * get_gtk_widget ();
+};
+
+EXPORT Lyrics aud_plugin_instance;
+
+const char * const Lyrics::defaults[] = {
+    "remote-source", "lyrics.ovh",
+    "enable-file-provider", "TRUE",
+    "enable-cache", "TRUE",
+    "split-title-on-chars", "FALSE",
+    "split-on-chars", "-",
+    "truncate-fields-on-chars", "FALSE",
+    "truncate-on-chars", "|",
+    "use-embedded", "TRUE",
+    nullptr
+};
+
+const PreferencesWidget Lyrics::widgets[] = {
+    WidgetLabel (N_("<b>General</b>")),
+    WidgetCheck (N_("Split title into artist and title on chars"),
+        WidgetBool (CFG_SECTION, "split-title-on-chars")),
+    WidgetTable ({{split_elements}}, WIDGET_CHILD),
+    WidgetLabel (N_("<b>Sources</b>")),
+    WidgetCheck (N_("Use embedded lyrics (from Lyrics tag)"),
+        WidgetBool (CFG_SECTION, "use-embedded")),
+    WidgetCombo (N_("Fetch lyrics from internet:"),
+        WidgetString (CFG_SECTION, "remote-source"),
+        {{remote_sources}}),
+    WidgetCheck (N_("Store fetched lyrics in local cache"),
+        WidgetBool (CFG_SECTION, "enable-cache")),
+    WidgetLabel (N_("<b>Local Storage</b>")),
+    WidgetCheck (N_("Load lyric files (.lrc) from local storage"),
+        WidgetBool (CFG_SECTION, "enable-file-provider"))
+};
```

Comment:
libaudgui does not show all widgets for some reason. This needs further verification.

Reply:
It looks like libaudgui does not support nested `WidgetTable`. See `fill_table()` in prefs-widget.cc.